js/src/Mark.ts

```diff
@@ -208,6 +208,28 @@ export abstract class Mark extends widgets.WidgetView {
         return this.colors[index % len];
     }
 
+    get_mark_color(data, index) {
```

Comment:
Those methods were in `ScatterBase.ts`, they will be very useful for other marks as well, not only for `Scatter`, so let's move them to the base class